server/CMakeLists.txt

```diff
@@ -1,22 +1,31 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 2.6)
 project(ins_server)
 
+set(CMAKE_C)
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
+
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
 
+find_package (LibXml2)
+
 set(SOURCE_FILES
     src/main.cpp
     src/ins_service.cpp
     src/data_store.cpp
     src/localization.cpp
     src/lib_wrapper.cpp
+    src/WifiNode.c
+    src/WifiAccessPointLocalConfig.c
  )
 
 include_directories(
     ${CMAKE_FIND_ROOT_PATH}/include
     ${CMAKE_CURRENT_SOURCE_DIR}/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/include/external )
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/external)
+
+link_directories(${CMAKE_CURRENT_SOURCE_DIR}/libxml2)
 
-link_directories(${CMAKE_FIND_ROOT_PATH}/lib)
+link_directories(${CMAKE_FIND_ROOT_PATH}/libxml2)
 add_executable(ins_server ${SOURCE_FILES})
-target_link_libraries(ins_server pistache.a sqlite3.a dl)
+target_link_libraries(ins_server pistache.a sqlite3.a dl m ${LIBXML2_LIBRARIES})
```

Comment:
> m ${LIBXML2_LIBRARIES}

Where is the `LIBXML2_LIBRARIES` set and what does that `m` do in the front of it?

Reply:
it is set from the fetch package command. The m is a math library.

Reply:
The ${LIBXML2_INCLUDE_DIR} actually worked for the main build and the test build. So I have removed the libxml2 header path from external/.
I think we can make it work for the spdlog library too with something like ${SPDLOG_INCLUDE_DIR} and ${SPDLOG_LIBRARIES}. @samup4web what do you think about that?